packages/context-editor/src/schemas/math.ts

```diff
@@ -1,4 +1,41 @@
-import type { DOMOutputSpec, MarkSpec, NodeSpec } from "prosemirror-model";
+import type { Node, NodeSpec } from "prosemirror-model";
+
+import { renderToString } from "katex";
+
+import { tryCatch } from "utils/try-catch";
+
+const renderMath = (node: Node, type: "math-inline" | "math-display") => {
+	const [err, renderedKatex] = tryCatch(() =>
+		renderToString(node.textContent, { output: "mathml" })
+	);
+
+	const content = err ? `<span class="parse-error">(math error)</span>` : renderedKatex;
+
+	// this is not nice, i would like to avoid manually calling `document.createElement`
+	// as we now need to keep track of setting `global.document` when this is called on the server
+	// forturnately, we only really server render the HTML when importing Legacy text content
+	// could be easily solved if `toDOM` was passed the document object, but ProseMirror does not seem to intend providing this
+	// https://discuss.prosemirror.net/t/getting-a-hold-of-the-document-used-by-prosemirror-from-todom/8392
+	if (!global.document) {
+		throw new Error(
+			"document not found. Trying to serialize math in a non-browser environment. To do this, set `global.document` to eg a `JSDOM` document before serializing."
+		);
+	}
```

Comment:
this is very annoying, if you know of a better way to have prosemirror convert a raw html string to a node let me know!